File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.js
function solution(numbers, direction) {
    if(direction === "right"){
        numbers.unshift(numbers.pop());
    }else{
        numbers.push(numbers.shift())
    }
    return numbers;
}
// function solution(numbers, direction) {
//     var answer = [];
//     if(direction == 'right'){
//         answer.push(numbers[numbers.length-1]);
//     } 
//     for(let i=0; i<numbers.length-1; i++){
//         if(direction == 'right'){
//             answer.push(numbers[i]);
//         } else if(direction == 'left'){
//             answer.push(numbers[i+1]);
//         }
//     }
//     if (direction == 'left'){
//         answer.push(numbers[0]);
//     }
//     return answer;
// }

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.py
def solution(numbers, direction):
    answer = []
    length = len(numbers)
    if direction == 'right':
        answer.append(numbers[length-1])
        for i in range(0, len(numbers)-1):
            answer.append(numbers[i])
    elif direction == 'left':
        for j in range(1, len(numbers)):
            answer.append(numbers[j])
        answer.append(numbers[0])
    return answer